mitiq/cdr/cdr_execution.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""API for using Clifford Data Regression (CDR) error mitigation."""
+
+from typing import List, Union, Callable, Sequence
+
+import numpy as np
+from scipy.optimize import curve_fit
+
+from cirq.circuits import Circuit
+
+from mitiq.cdr.clifford_training_data import generate_training_circuits
+from mitiq.cdr.data_regression import (
+    construct_training_data_floats,
+    construct_circuit_data_floats,
+    linear_fit_function,
+)
+from mitiq.zne.scaling import fold_gates_at_random
+
+
+# TODO: Allow for any QPROGRAM, not just a cirq.Circuit.
+def execute_with_CDR(
```

Comment:
```suggestion
def execute_with_cdr(
```

For style convention and for consistency with other Mitiq `execute_with_something` functions.